src/puppetlabs/i18n/core.clj

```diff
@@ -61,14 +61,16 @@
     (map #(-> % slurp read-string (assoc :source %) check-locales check-and-normalize-packages)
          (info-files))))
 
-(defn info-map
-  "Turn the result of infos into a map mapping the package name to locales
-  and bundle name.
-
-  To facilitate testing, we allow multiple infos with the same :package as
-  long as they agree on the :bundle. The result of such a setup is that
-  the :locales for such a package are the union of all the locales from
-  those info files"
+(def string-length-comparator
+  (reify java.util.Comparator
+    (compare [_ lhs rhs]
+      (clojure.core/compare (count lhs) (count rhs)))))
```

Comment:
What we need is a comparator that sorts keys by descending length, since when we look up entries from `info-map`, we want to prefer longer matches. The compare should be something like `(clojure.core/compare (count rhs) (count lhs))`